### DISCUSSION TITLE
How to programatically check if the given field rules are valid

### DISCUSSION THREAD

**Discussion by ajaybeniwal** [react-hook-form/react-hook-form#2089] [Q&A]
I am using Antd Components wrapped inside Controller component , I have a use case where i have to check the validatity of rules on field inside onChangeEvent to enable some other indicator in the form , so can i check programatically if all rules are met on control inside the onChangeEvent

**Comment by bluebill1049** [comment]
trigger? or setError?

**Reply by bluebill1049** [reply]
you can use setValue to trigger vlaidation as well

**Reply by ajaybeniwal** [reply]
so you mean to say isFieldValid = await trigger("lastName") , but since this process looks like asynchronous will it have any performance impact if i do it onChangeEvent ?

**Reply by bluebill1049** [reply]
it's fine, it's async because you can have async validation.
